### PR TITLE
Add `SpectrumDataset.create()`

### PR DESCRIPTION
**Pull request by registerrier**
This is a first PR to introduce creation of empty datasets for data reduction processing. 

The `counts` and `background` are empty. the `aeff` is 0 but `edisp` is diagonal.
`livetime` is set to 0 as well.

### CODE REVIEW

File: gammapy/spectrum/dataset.py [hunk 1]
```diff
@@ -359,6 +359,34 @@ def plot_residuals(self, method="diff", ax=None, **kwargs):
...
+        mask_safe = np.ones_like(counts.data.data, 'bool')
+        livetime = 0*u.s
+
+        return SpectrumDataset(counts=counts,
+                               aeff=aeff,
+                               edisp=edisp,
+                               mask_safe=mask_safe,
+                               background=background,
+                               livetime=livetime,
+                               )
```
Comment (adonath) [inline gammapy/spectrum/dataset.py]: Maybe also create and add an empty `GTI` table?

File: gammapy/spectrum/dataset.py [hunk 2]
```diff
@@ -359,6 +359,38 @@ def plot_residuals(self, method="diff", ax=None, **kwargs):
...
+        e_reco : `~astropy.units.Quantity`
+            edges of counts vector
+        e_true : `~astropy.units.Quantity`
+            edges of effective area table. If not set use reco energy values. Default : None
+        reference_time : `~astropy.time.Time`
+            reference time of the dataset, Default is "2000-01-01"
+        """
+        counts = CountsSpectrum(e_reco[:-1], e_reco[1:])
+        background = CountsSpectrum(e_reco[:-1], e_reco[1:])
+        aeff = EffectiveAreaTable(e_true[:-1],e_true[1:], np.zeros(e_true[:-1].shape)*u.m**2)
```
Comment (adonath) [inline gammapy/spectrum/dataset.py]: As `e_true` is optional, I guess there is one line missing:
```
if e_true is None:
    e_true = e_reco
```

File: gammapy/spectrum/dataset.py [hunk 3]
```diff
@@ -359,6 +359,38 @@ def plot_residuals(self, method="diff", ax=None, **kwargs):
...
+        edisp = EnergyDispersion.from_diagonal_response(e_true, e_reco)
+        mask_safe = np.ones_like(counts.data.data, 'bool')
```
Comment (adonath) [inline gammapy/spectrum/dataset.py]: Note that `CountsSpectrum.data` is now a Numpy array and not an `NDDataArray`anymore, so writing `counts.data` is sufficient.

File: gammapy/spectrum/dataset.py (hunk 2 above)
Comment (registerrier) [reply gammapy/spectrum/dataset.py]: Corrected

File: gammapy/spectrum/dataset.py (hunk 3 above)
Comment (registerrier) [reply gammapy/spectrum/dataset.py]: Right, corrected.

**Pull request merged**